profiles summary: tweaks

yurique opened the pull request:
Changing the text for SMB delivery range:
* if any of the settings are changed (min, max, range) - the whole description is displayed
* instead of three separate texts - display one formatted text like `0.5-0.7 range 5.8 mmol/L`
* if min=max, display just one number

---

Duration is formatted: * `1h 3m`.

---

Use the number formatter for decimal values, to prevent it from displaying ugly floating point numbers like `1.7898989898989898989`

---

Refactoring to use `if something { Text(something) }` instead of `something ? Text(something) : nil` - adding `nil`s seemed to affect the layout (like adding extra pixels of padding)

Before/after compared to `dev`:

---

<img width="350" alt="image" src="https://github.com/user-attachments/assets/3a411dfb-7b69-4883-b4d3-31a73e852f46" />

<img width="350" alt="image" src="https://github.com/user-attachments/assets/95fa928e-e966-4189-87ec-bc77ae975a5b" />

---

<img width="350" alt="image" src="https://github.com/user-attachments/assets/d43239c6-8376-47f6-9ce6-077939bb46b0" />

<img width="350" alt="image" src="https://github.com/user-attachments/assets/329ebc7b-20d0-497e-99f5-13610fefbe87" />

Jon-b-m commented:
After you're done please make a screenshot before after (comparing with latest dev at that time) to illustrate what you want to accomplish. As mentioned elsewhere there already is a PR in progress, why I suggest looking at this also, to avoid extra work.

Jon-b-m commented:
The PR aforementioned is merged into dev now.

yurique commented on FreeAPS/Sources/Modules/OverrideProfilesConfig/View/OverrideProfilesRootView.swift:
```diff
@@ -703,19 +701,25 @@ extension OverrideProfilesConfig {
-                            if ((aisf.smbDeliveryRatioMin ?? 0.5) as Decimal) != standard
-                                .smbDeliveryRatioMin || ((aisf.smbDeliveryRatioMax ?? 0.5) as Decimal) != standard
-                                .smbDeliveryRatioMax
+                            let smbDeliveryRatioMin = aisf.smbDeliveryRatioMin as? Decimal ?? standard.smbDeliveryRatioMin
```
if any of the three settings are changed - display the whole text (`SMB ratio: 0.3-0.7 range 5.6 mmol/L`)
if min=max - only display one value (`SMB ratio: 0.5`)

I wonder if `SMB ratio: 0.3-0.7 over 5.6 mmol/L` ("over" instead of "range") is better.

Jon-b-m replied:
No, because ”over” can mean several things. Better to be precise here.

yurique commented on FreeAPS/Sources/Modules/OverrideProfilesConfig/View/OverrideProfilesRootView.swift:
```diff
@@ -689,10 +689,8 @@ extension OverrideProfilesConfig {
-                                decimal(decimal: aisf.autoisf_min, setting: standard.autoisf_min, label: "Min: ")
-                                decimal(decimal: aisf.autoisf_max, setting: standard.autoisf_max, label: "Max: ")
-                            }
+                            decimal(decimal: aisf.autoisf_min, setting: standard.autoisf_min, label: "Min: ")
+                            decimal(decimal: aisf.autoisf_max, setting: standard.autoisf_max, label: "Max: ")
```
it fills like it should be able to combine these two into one text with a range (`0.7-2.0`) - but I can't decide which label to use in that case

Jon-b-m replied:
No need to make more complex every body knows what min and max means. It’s clear and precise already imo.

yurique commented on FreeAPS/Sources/Modules/OverrideProfilesConfig/View/OverrideProfilesRootView.swift:
```diff
@@ -689,10 +689,8 @@ extension OverrideProfilesConfig {
                             bool(bool: aisf.ketoProtect, setting: standard.ketoProtect, label: "Keto: ")
                             bool(bool: aisf.use_B30, setting: standard.use_B30, label: "B30: ")
 
-                            LazyHStack(spacing: 5) {
```
`LazyHStack` was adding large vertical padding for some reason

yurique commented:
currently it looks like this:
<img width="403" alt="image" src="https://github.com/user-attachments/assets/16ae6f41-195b-40bd-82b8-19b1a4f2169c" />

yurique commented:
current before/after (still not perfect):

<img width="400" alt="image" src="https://github.com/user-attachments/assets/2c1e5f03-4f48-4f15-8c95-c616f535f4c6" />

<img width="400" alt="image" src="https://github.com/user-attachments/assets/4d1ff4e5-ea1d-4c5a-89b4-4fda7fed8bbb" />

Jon-b-m commented:
OK. I think it looks better before

yurique commented:
How about this? (all tags are at the top, first line is white)

<img width="400" alt="image" src="https://github.com/user-attachments/assets/6675908f-8079-471e-a3d0-ee02bdf5c192" />

yurique commented:
with `Max IOB` on the first line 

<img width="400" alt="image" src="https://github.com/user-attachments/assets/66a715b1-4744-4c61-8041-53d75f79a801" />

yurique commented:
`Scheduled SMBs` as a tag as well

<img width="400" alt="image" src="https://github.com/user-attachments/assets/9952f706-de73-4209-8823-55f249264aad" />

yurique commented:
some padding under the tags line

<img width="400" alt="image" src="https://github.com/user-attachments/assets/b7c7d270-e751-44b0-b082-fc4a14adc3fb" />

Jon-b-m commented:
It's a bit distracting having big AISF settings tags in first line. And I think it's a peculiar prioritising. The current dev/main has the sections in order of priority, not just by looks. 
The SMBs scheduled is also peculiar as a tag, because it's not exactly clear what the tag is meant to illustrate.  Does the red colour mean it's not scheduled? I therefore don't think that is very functional or helpful.

Jon-b-m commented:
I converted to draft, because it seems this PR is still in an idea/draft stage of development.

yurique commented:
> I find it a bit distracting having big AISF settings tags in first line.

I'd argue those settings are the most important when they are overridden - since the turn parts of the algorithm *completely* on or off:
* if I disable `accel` for the night - that's the biggest thing of the profile
* if I enable `B30` - that seems to be pretty important as well
* if autoISF is enabled/disabled in the override - it changes the algorithm completely
* disabled SMBs are important as well

> The SMBs scheduled is also peculiar as a tag, because it's not exactly clear what the tag is meant to illustrate

Maybe it could be a gray tag instead of red? This one is probably less important than "SMBs are off", but still important - for those who use it. 
I tried making it a tag so that it's in the same place as "SMBs are off", but it can go back to where it was. 
Or maybe it could have a separate line - with scheduled hours displayed?

---

I don't have a lot of real profiles on my phone, so it's harder for me to judge - would be nice to see what it looks with real profiles that people have. But I'd imagine most of the profiles will not have all the tags displayed - just one or two, and those will be important parts of the override.

---

On the other hand, I'm not really "married" to the idea of moving the tags to the top :) – I can move them back to where they were. But it's harder to prevent random un-pretty margins and gaps when tags can appear on multiple rows:

<img width="400" alt="image" src="https://github.com/user-attachments/assets/3bbdec26-b273-43ed-a886-711640c5b027" />

yurique commented:
reverted the tags to previous positions

before (current `dev`) / after (this branch):

<img width="400" alt="image" src="https://github.com/user-attachments/assets/0fc14239-ff17-4503-96e1-1443b397e985" />

<img width="400" alt="image" src="https://github.com/user-attachments/assets/531ac671-ce30-4bea-98a6-d86c6ccd177d" />

yurique commented:
another direction to explore - everything is a tag:

<img width="400" alt="image" src="https://github.com/user-attachments/assets/76434bcd-2fbb-4277-873d-ad0253a6837d" />

<img width="400" alt="image" src="https://github.com/user-attachments/assets/da2e6795-782a-4c70-b852-dd2ef723dd29" />

Looks... maybe interesting, I guess? :) 

Something is adding random 1-2 pixel of extra space between some rows, and it's annoying. And I don't see why.

yurique commented:
current status:

<img width="400" alt="image" src="https://github.com/user-attachments/assets/9e14fd60-99e8-433e-b8d1-037d6c4d57ed" />

yurique commented:
fixed text misalignment


<img width="400" alt="image" src="https://github.com/user-attachments/assets/54e98b6c-aca7-4f46-ad70-d227200d7881" />

yurique commented:
more padding between tags, formatted duration:

<img width="400" alt="image" src="https://github.com/user-attachments/assets/0cb06b54-7e6d-4321-bb29-3774b016da4e" />

yurique commented:
a bit less padding between adjacent tags

<img width="400" alt="image" src="https://github.com/user-attachments/assets/edd8fe5e-86ac-4325-b78f-a2915f1d1cd6" />

yurique commented:
updated the description, the up-to-date screenshots are in there

Jon-b-m commented:
Thank you for the screenshots. And nice to finally have a description of what you’re actually intending with this PR…

Overall this suggestion still looks incomplete for the following reasons:

Profile 2,3,4 and 6 all look different from one another. There is no consistency and the spacing and padding isn’t the same. 

The added verbose time function isn’t needed in SwiftUI. There is a much cleaner ways of formatting the time. 

The ViewBuilder change here has no net effect (different UI).

yurique commented:
> Profile 2,3,4 and 6 all look different from one another.

Nothing is changed about this inconsistency in this PR - it's exactly the same as in current `dev` (the screenshots on the left are made in the `dev` branch).

> The ViewBuilder change here has no net effect (different UI).

👍 in that case it's not necessary, reverted

> There are cleaner ways of formatting the time.

replaced with `DateComponentsFormatter` (hours are always `h`, minutes are always `m`)

Jon-b-m commented:
To add to the list it's unfortunately also lacking localization. And in current form it can't be read and translated in Crowdin as you're using string  interpolation without  the necessary wrapping

Jon-b-m commented:
You're also mentioning padding and spacing in this PR, however hone of that is adressed?

yurique commented:
It will have to be a separate PR.

Jon-b-m commented:
I mentioned some days ago that I already had a local PR fixing everything requested with minimal code. But I wanted you to still have a change doing these changes yourself since you talked so much in private DMs and here about this PR. But it makes no sense merging something which isn’t fixing half of the issues you wanted to fix and asked me how to fix?

yurique commented:
At this point I really don't know what we're talking about. 

I wanted to make some adjustments. I implemented them. 
I did want to deal with padding issues as well. And I implemented the changes for that here initially, then iterated a few times with different suggestions/ideas.
Eventually, you told me I wasn't using something from `dev` which would make 80% of my changes unnecessary – I wasn't able to find what exactly I wasn't using, and I knew you already had a simpler fix for the padding issues – so it only made sense for me to remove the over-complicated changes in order to make the PR about the other things only, so we could stop going in circles about the padding. 

If the changes proposed here are wrong or not good enough - we can simply close this PR. No harm in doing that.

Jon-b-m commented:
Ok. I’ll close and push my old Pr.
